src/org/labkey/test/components/domain/DomainFormPanel.java

```diff
@@ -317,19 +317,31 @@ public String getPanelAlertText()
         return getPanelAlertWebElement().getText();
     }
 
+    public String getPanelAlertText(int index)
+    {
+        return getPanelAlertWebElement(index).getText();
+    }
+
     /**
      * There may be an element in the alert that a test will need to interact with so return the alert element and let
      * the test find the control it needs.
      * @return The div wrapping the alert in the panel, null otherwise.
      */
     public WebElement getPanelAlertWebElement()
     {
+        return getPanelAlertWebElement(0);
+    }
+
+    public WebElement getPanelAlertWebElement(int index)
+    {
+
         getWrapper().waitFor(()-> BootstrapLocators.infoBanner.existsIn(getDriver()),
                 "the info alert did not appear as expected", 1000);
 
         // It would be better to not return a raw WebElement but who knows what the future holds, different alerts
         // may show different controls.
-        return BootstrapLocators.infoBanner.existsIn(getDriver()) ? BootstrapLocators.infoBanner.findElement(getDriver()) : null;
+        return BootstrapLocators.infoBanner.existsIn(getDriver()) ?
+                BootstrapLocators.infoBanner.findElements(getDriver()).get(index) : null;
```

Comment:
Should this not be scoped to the panel? Also, you can use `findOptionalElement` to avoid finding the banner twice.
```suggestion
        return BootstrapLocators.infoBanner.index(index).findOptionalElement(this).orElse(null);
```

Reply:
Ah, that's nice.  I'll do that.